openml/_api_calls.py

```diff
@@ -198,15 +201,17 @@ def _send_request(
                 if isinstance(e, OpenMLServerException):
                     if e.code != 107:
                         # 107 is a database connection error - only then do retries
-                        raise
+                        raise e
```

Comment:
Could you please also check for the `500 internal server error` and try again on that?

Reply:
Side node, I believe `raise` and `raise e` are equivalent here since `e` is the currently handled exception.

Reply:
@mfeurer I edited the structure a bit to allow further scaling of such retries if needed. Hope this is okay.

@PGijsbers didn't know that, thanks!